crates/router/src/core/revenue_recovery/types.rs

```diff
@@ -563,18 +645,94 @@ impl Action {
         match response {
             Ok(_payment_data) => match payment_attempt.status.foreign_into() {
                 RevenueRecoveryPaymentsAttemptStatus::Succeeded => {
+                    let payment_intent = db
```

Comment:
We can't read payment related database calls, which against the modularity, refer with @Aprabhat19